core/batcher.go

```diff
@@ -0,0 +1,348 @@
+		span.SetStatus(codes.Error, "marshalling batch response failed")
+
+		return nil, fmt.Errorf("failed marshalling batch response %s: %w", batchID, err)
+	}
+
+	return data, nil
+}
+
+const skiPattern = "^[0-9a-f]+$"
+
+func validateBatcherSKI(batcherSKI string) error {
+	ok, err := regexp.MatchString(skiPattern, batcherSKI)
+	if err != nil {
+		return fmt.Errorf("failed to compile regexp pattern ski: %w", err)
+	}
+
+	if !ok {
+		return fmt.Errorf("invalid batcher SKI %s: does not match regex pattern '%s'", batcherSKI, skiPattern)
+	}
+
+	return nil
+}
+
+func NewBatcher(stub shim.ChaincodeStubInterface, cfgBytes []byte, cc *ChainCode, tracingHandler *telemetry.TracingHandler) (Batcher, error) {
+	contractCfg, err := config.ContractConfigFromBytes(cfgBytes)
+	if err != nil {
+		return Batcher{}, fmt.Errorf("LoadConfig: contract config from bytes: %w", err)
+	}
+
+	batchCacheStub := cachestub.NewBatchCacheStub(stub)
+
+	batcherSKI := contractCfg.GetBatcherSKI()
+	err = validateBatcherSKI(batcherSKI)
+	if err != nil {
+		return Batcher{}, fmt.Errorf("failed to validate batcher SKI: %w", err)
+	}
+
+	return Batcher{
+		batchCacheStub: batchCacheStub,
+		cfgBytes:       cfgBytes,
+		ski:            batcherSKI,
+		cc:             cc,
+		tracingHandler: tracingHandler,
+	}, nil
+}
+
+func (b *Batcher) ValidateCreator(creatorSKI [32]byte, hashedCert [32]byte) error {
+	skiBytes, err := hex.DecodeString(b.ski)
+	if err != nil {
+		return fmt.Errorf("failed to decode hex batcherSKI: %w", err)
+	}
+
+	err = hlfcreator.ValidateSKI(skiBytes, creatorSKI, hashedCert)
+	if err != nil {
+		return fmt.Errorf("unauthorized: batcherSKI is not equal creatorSKI and hashedCert: %w", err)
+	}
+
+	return nil
+}
+
+func (b *Batcher) HandleBatch(
+	traceCtx telemetry.TraceContext,
+	requests []BatcherRequest,
+) (
+	*proto.BatchResponse,
+	*proto.BatchEvent,
+	error,
+) {
+	traceCtx, span := b.tracingHandler.StartNewSpan(traceCtx, "Batcher.HandleBatch")
+	defer span.End()
+
+	batchResponse := proto.BatchResponse{}
+	batchEvent := proto.BatchEvent{}
+
+	for _, request := range requests {
+		var (
+			txResponse *proto.TxResponse
+			txEvent    *proto.BatchTxEvent
+		)
+		switch request.BatcherRequestType {
+		case TxBatcherRequestType:
+			txResponse, txEvent = b.HandleTxRequest(traceCtx, request, b.batchCacheStub, b.cfgBytes)
+		default:
+			errorMessage := fmt.Sprintf("unsupported batcher request type %s request.BatcherRequestID %s", request.BatcherRequestType, request.BatcherRequestID)
+			txResponse, txEvent = returnError(span, request, errorMessage)
+		}
+		batchResponse.TxResponses = append(batchResponse.TxResponses, txResponse)
+		batchEvent.Events = append(batchEvent.Events, txEvent)
+	}
+
+	if err := b.batchCacheStub.Commit(); err != nil {
+		return nil, nil, fmt.Errorf("failed to commit changes using batchCacheStub: %w", err)
+	}
+
+	return &batchResponse, &batchEvent, nil
+}
+
+func (b *Batcher) validatedTxSenderMethodAndArgs(
+	stub *cachestub.BatchCacheStub,
```

Comment:
The name *cachestub.BatchCacheStub does not conform to Go style due to duplication of the package name in the type name. In Go, it is recommended to avoid redundancy by not repeating the package name in the type name.

Variants:
```
package cachestub

type BatchCache struct {
}
```

or

```
package cachestub

type Batch struct {
}
```

`cachestub.BatchCache` or `cachestub.Batch`

like `bytes.Buffer` etc.

https://google.github.io/styleguide/go/decisions#repetitive-with-package

Reply:
I created the issue for that https://github.com/anoideaopen/foundation/issues/47 because this changes independent with this PR